pkg/apis/componentconfig/validation/validation.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+)
+
+// ValidateKubeProxyConfiguration validates the configuration of kube-proxy
+func ValidateKubeProxyConfiguration(config *componentconfig.KubeProxyConfiguration) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	newPath := field.NewPath("KubeProxyConfiguration")
+
+	allErrs = append(allErrs, validateKubeProxyIPTablesConfiguration(config.IPTables, newPath.Child("KubeProxyIPTablesConfiguration"))...)
+	allErrs = append(allErrs, validateKubeProxyConntrackConfiguration(config.Conntrack, newPath.Child("KubeProxyConntrackConfiguration"))...)
+	allErrs = append(allErrs, validateProxyMode(config.Mode, newPath.Child("Mode"))...)
+
```

Comment:
See: 
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/49087/pull-kubernetes-e2e-kops-aws/35881/nodelog?junit=junit_runner.xml&wrap=on

In kube-proxy.log:

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/49087/pull-kubernetes-e2e-kops-aws/35881/artifacts/ec2-34-212-154-98.us-west-2.compute.amazonaws.com/kube-proxy.log

```
Flag --resource-container has been deprecated, This feature will be removed in a later release.
W0719 06:07:49.070548       5 server.go:191] WARNING: all flags other than --config, --write-config-to, and --cleanup-iptables are deprecated. Please begin using a config file ASAP.
F0719 06:07:49.070721       5 helpers.go:119] KubeProxyConfiguration.KubeProxyIPTablesConfiguration.MinSyncPeriod: Invalid value: v1.Duration{Duration:0}: must be greater than 0
KubeProxyConfiguration.KubeProxyConntrackConfiguration.Min: Invalid value: 131072: cannot be greater than KubeProxyConntrackConfiguration.Max
```

kube-proxy fails to start due to the validation. But I don't know where to see the config of kube-proxy.